Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm

```diff
@@ -591,6 +602,127 @@ function log(msg)
+        }];
+        Util::run(&done);
+        auto notificationMessageHandler = adoptNS([[NotificationScriptMessageHandler alloc] init]);
+        auto testMessageHandler = adoptNS([[TestMessageHandler alloc] init]);
+
+        __block bool ready = false;
+        [testMessageHandler addMessage:@"Ready" withHandler:^{
+            ready = true;
+        }];
+        __block bool gotExpectedMessage = false;
+        [testMessageHandler addMessage:@"Received: test aesgcm payload" withHandler:^{
+            gotExpectedMessage = true;
+        }];
+
+        auto webViewConfiguration = adoptNS([WKWebViewConfiguration new]);
+        [webViewConfiguration.get().preferences _setPushAPIEnabled:YES];
+        [webViewConfiguration setWebsiteDataStore:dataStore.get()];
+        enableNotifications(webViewConfiguration.get());
+        [webViewConfiguration.get().userContentController addScriptMessageHandler:notificationMessageHandler.get() name:@"note"];
+        [webViewConfiguration.get().userContentController addScriptMessageHandler:testMessageHandler.get() name:@"test"];
+
+        auto notificationProvider = makeUnique<TestWebKitAPI::TestNotificationProvider>(Vector<WKNotificationManagerRef> { [[webViewConfiguration processPool] _notificationManagerForTesting], WKNotificationManagerGetSharedServiceWorkerNotificationManager() });
+        notificationProvider->setPermission("https://example.com"_s, true);
+
+        auto webView = adoptNS([[WKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:webViewConfiguration.get()]);
+        auto uiDelegate = adoptNS([NotificationPermissionDelegate new]);
+        auto navigationDelegate = adoptNS([TestNavigationDelegate new]);
+        navigationDelegate.get().didReceiveAuthenticationChallenge = ^(WKWebView *, NSURLAuthenticationChallenge *challenge, void (^completionHandler)(NSURLSessionAuthChallengeDisposition, NSURLCredential *)) {
+            completionHandler(NSURLSessionAuthChallengeUseCredential, [NSURLCredential credentialForTrust:challenge.protectionSpace.serverTrust]);
+        };
+        [webView setUIDelegate:uiDelegate.get()];
+        [webView setNavigationDelegate:navigationDelegate.get()];
+        [webView loadRequest:[NSURLRequest requestWithURL:exampleURL]];
+        TestWebKitAPI::Util::run(&ready);
+
+        auto testPush = ^(bool expectedMessages){
+            gotExpectedMessage = false;
+            auto message = WebPushDTest::injectPushMessage(aesGCMPushData(), exampleURL, dataStore.get(), expectedMessages);
+
+            __block bool pushMessageProcessed = false;
+            __block bool pushMessageProcessedResult = false;
+            [dataStore _processPushMessage:message.get() completionHandler:^(bool result) {
+                pushMessageProcessedResult = result;
+                pushMessageProcessed = true;
+            }];
+            if (!expectedMessages) {
+                usleep(100000);
+                TestWebKitAPI::Util::spinRunLoop(100);
```

Comment:
I think you can just use `TestWebKitAPI::Util::sleep(0.1)` instead of a usleep() + spinRunLoop()